Src/BootCamp.Chapter/Lesson3.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+
+
+namespace BootCamp.Chapter
+{    public class Lesson3
+    {   public static void Demo()
+        {
+            for (int i = 0; i < 2; i++)
+            {
+                Console.Clear();
+                GatherData(out string firstName, out string lastName, out int age, out float weight, out float height);
+                DisplayOutput(firstName, lastName, age, weight, height);
+            }
+        }
+        public static string GetStringInput(string prompt)
+        {
+            Console.WriteLine(prompt);
+            string value = Console.ReadLine();
+            
+            if (Empty(value))
+            {
+                Console.Write("Name cannot be empty.");
+                return "-";
+            }
+
+            //default case
+            return value;
+        }
+        public static int GetIntInput(string prompt)
+        {
+            Console.WriteLine(prompt);
+            string value = Console.ReadLine();
+
+            if (Empty(value))
+            {
+                return 0;
+            }
+
+            if (int.TryParse(value, out int parsedInt))
+            {
+                return parsedInt;
+            }
+
+            //default case
+            BadInput(value);
+            return -1;
+            
+        }
+        public static float GetFloatInput(string prompt)
+        {
+            Console.WriteLine(prompt);
+            string value = Console.ReadLine();
+
+            if (Empty(value))
+            {
+                return 0;
+            }
+
+            if (float.TryParse(value, out float parsedfloat))
+            {
+                return parsedfloat;
+            }
+
+            //default case
+            BadInput(value);
+            return -1;
+
+        }
+        // Input function
+        private static void GatherData(out string firstName, out string lastName, out int age, out float weight, out float height)
+        {
+            firstName = GetStringInput("Hello, What is your First name?");
+            lastName = GetStringInput("Hello, What is your Last name?");
+
+            age = GetIntInput("How old are you?");
+            weight = GetFloatInput("How much do you weigh in kilograms?");
+            height = GetFloatInput("How tall are you?") / 100;
+            if (height <= 0)
+            {
+                Console.WriteLine("Press [Enter] to continue");
+                Console.ReadKey();
+            }
+        }
+        // Output function
+        public static void DisplayOutput(string first, string last, int age, float weight, float height)
+        {
+            Console.Clear();
+            Console.WriteLine(first + " " + last + " is " + age + " years old; his weight is " + weight + " kg and his height is " + (height * 100) + " cm.");
+            Console.WriteLine(first + "'s Body Mass Index is " + CalcBmi(weight, height) + ".");
+            Console.WriteLine("\n\n\n\n\nPress Enter");
```

Comment:
If you make use of NewLine characters more often you could think about creating a function like WriteNewLines(int count).

Reply:
agreed, but i felt that was overkill for this instance.